src/main/java/ch/bfh/bti7081/s2018/green/models/entities/Program.java

```diff
@@ -0,0 +1,77 @@
+package ch.bfh.bti7081.s2018.green.models.entities;
+
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Period;
+
+public class Program {
+
+    private Therapy therapy;
+
+    private LocalDate start;
+
+    private LocalDate stop;
+
+    private Period frequence;
+
+    private RecurringEvent firstEvent;
+
+    public Program(Therapy therapy, LocalDate start, LocalDate stop, Period frequence, RecurringEvent firstEvent) throws IllegalArgumentException {
+        this.therapy = therapy;
+        this.start = start;
+        this.stop = stop;
+        this.frequence = frequence;
+        this.firstEvent = createRecuringEvents(firstEvent);
+    }
+
+    public Therapy getTherapy() {
+        return therapy;
+    }
+
+    public LocalDate getStart() {
+        return start;
+    }
+
+    public LocalDate getStop() {
+        return stop;
+    }
+
+    public Period getFrequence() {
+        return frequence;
+    }
+
+    public RecurringEvent getFirstEvent() {
+        return firstEvent;
+    }
+
+    private RecurringEvent createRecuringEvents(RecurringEvent firstEvent) throws IllegalArgumentException {
```

Comment:
I recommed to improve the variable names as follows:
- previousEvent instead of lastEvent (otherwise one might think it refers to the last event of the series)
- currentEvent instead of actualEvent
- "recurring" should consistently be written with double "r"

What's more I would rename the exception message to:
"Recurring event's start date doesn't correspond to the start date of the program"